### PR TITLE
Remove `toolz.itertoolz.concatv`

### PR DESCRIPTION
**Pull request by kenodegard**
### Description

Opting to use argument unpacking instead of `toolz.itertoolz.concatv`. All of the uses of `concatv` were immediately expanded into static types, so the generator `concatv` returned was unnecessary.

Alternative to #11769 

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [ ] ~Add / update outdated documentation?~

### CODE REVIEW

File: conda/core/link.py [hunk 1]
```diff
@@ -892,9 +898,9 @@ def _make_link_actions(transaction_context, package_info, target_prefix, request
         # else:
         #     application_entry_point_actions = ()
         #     application_softlink_actions = ()
-        # leased_paths = tuple(axn.leased_path_entry for axn in concatv(
-        #     application_entry_point_actions,
-        #     application_softlink_actions,
+        # leased_paths = tuple(axn.leased_path_entry for axn in (
+        #     *application_entry_point_actions,
+        #     *application_softlink_actions,
```
Comment (dholth) [inline conda/core/link.py:903]: delete comment?

File: conda/core/link.py [hunk 2]
```diff
@@ -905,13 +911,13 @@ def _make_link_actions(transaction_context, package_info, target_prefix, request
         #     register_private_env_actions = ()
 
         # the ordering here is significant
```
Comment (dholth) [inline conda/core/link.py:913]: oh yeah? YOU're significant

**Review by dholth — APPROVED** [review]

File: conda_env/env.py [hunk 3]
```diff
@@ -108,18 +102,24 @@ def from_environment(name, prefix, no_builds=False, ignore_channels=False, from_
...
+        ),
+        key=lambda x: x.name,
+    )
+
+    pip_precs = sorted(
+        (
+            *grouped_precs.get(PackageType.VIRTUAL_PYTHON_WHEEL, ()),
+            *grouped_precs.get(PackageType.VIRTUAL_PYTHON_EGG_MANAGEABLE, ()),
+            *grouped_precs.get(PackageType.VIRTUAL_PYTHON_EGG_UNMANAGEABLE, ()),
+            # *grouped_precs.get(PackageType.SHADOW_PYTHON_EGG_LINK, ()),
```
[the remark this answers is not in the document]
Comment (kenodegard) [reply conda_env/env.py:119]: I have no idea what `PackageType.SHADOW_PYTHON_EGG_LINK` refers to, so I'm worried potentially important information will get lost.

**Comment by dholth** [comment]
I'm OK with transforms like this one not including any refactors (that happen to be nearby the transform)

File: conda/core/index.py [hunk 4]
```diff
@@ -327,7 +327,7 @@ def push_record(record):
     # add feature records for the solver
     known_features = set()
     for rec in reduced_index.values():
-        known_features.update(concatv(rec.track_features, rec.features))
+        known_features.update((*rec.track_features, *rec.features))
```
Comment (dholth) [inline conda/core/index.py:330]: Good remembering it winds up with `((` and `))` on either side

**Review by dholth — APPROVED** [review]
LGTM

**Pull request merged**